src/Utilities/Compiler/Lightup/ITypeSymbolExtensions.cs

```diff
@@ -11,11 +11,16 @@ internal static class ITypeSymbolExtensions
             = LightupHelpers.CreateSymbolPropertyAccessor<ITypeSymbol, NullableAnnotation>(typeof(ITypeSymbol), nameof(NullableAnnotation), fallbackResult: Lightup.NullableAnnotation.None);
         private static readonly Func<ITypeSymbol, NullableAnnotation, ITypeSymbol> s_withNullableAnnotation
             = LightupHelpers.CreateSymbolWithPropertyAccessor<ITypeSymbol, NullableAnnotation>(typeof(ITypeSymbol), nameof(NullableAnnotation), fallbackResult: Lightup.NullableAnnotation.None);
+        private static readonly Func<ITypeParameterSymbol, NullableAnnotation> s_referenceTypeConstraintNullableAnnotation
+            = LightupHelpers.CreateSymbolPropertyAccessor<ITypeParameterSymbol, NullableAnnotation>(typeof(ITypeParameterSymbol), nameof(ReferenceTypeConstraintNullableAnnotation), fallbackResult: Lightup.NullableAnnotation.None);
 
         public static NullableAnnotation NullableAnnotation(this ITypeSymbol typeSymbol)
             => s_nullableAnnotation(typeSymbol);
 
         public static ITypeSymbol WithNullableAnnotation(this ITypeSymbol typeSymbol, NullableAnnotation nullableAnnotation)
             => s_withNullableAnnotation(typeSymbol, nullableAnnotation);
+
+        public static NullableAnnotation ReferenceTypeConstraintNullableAnnotation(this ITypeParameterSymbol typeParameterSymbol)
+            => s_referenceTypeConstraintNullableAnnotation(typeParameterSymbol);
```

Comment:
💡 Can you move this to `ITypeParameterSymbolExtensions` for the lightup layer?

Reply:
Moved to `ITypeParameterSymbolExtensions` and spotted/addressed more issues with detecting oblivious in type constraints. Thanks